slsa/provenance.go

```diff
@@ -0,0 +1,128 @@
+package slsa
+
+import (
+	"fmt"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+
+	"github.com/slsa-framework/slsa-github-generator/github"
+)
+
+// GithubHostedActionsBuilderID is the builder ID for Github hosted actions.
+const GithubHostedActionsBuilderID = "https://github.com/Attestations/GitHubHostedActions@v1"
```

Comment:
This will be set dynamically as per https://github.com/slsa-framework/slsa-github-generator-go/pull/17

Reply:
My understanding was that the builder ID identifies the builder at the granularity of the build system (i.e. Github hosted runners, Github self-hosted, Cloud Build hosted, Cloud build custom workers etc) and not necessarily identify the reusable workflow that was used.

If we use a URI for the reusable workflow, how might one differentiate between hosted versus self-hosted Github Actions for example? I can imagine scenarios where folks may want to verify the builder id to make sure that builds are run from their self hosted runners.

Reply:
Does it not make sense for the language-specific reusable workflows to set a URI for the workflow into the invocation instead? that feels more like the right place.

Reply:
> Does it not make sense for the language-specific reusable workflows to set a URI for the workflow into the invocation instead? that feels more like the right place.

definitely makes sense for the reusable workflow to set it, possibly via an API/CLI option?

Reply:
> My understanding was that the builder ID identifies the builder at the granularity of the build system (i.e. Github hosted runners, Github self-hosted, Cloud Build hosted, Cloud build custom workers etc) and not necessarily identify the reusable workflow that was used.
> 
> If we use a URI for the reusable workflow, how might one differentiate between hosted versus self-hosted Github Actions for example? I can imagine scenarios where folks may want to verify the builder id to make sure that builds are run from their self hosted runners.

The idea is to have a mapping between reusable workflows and whether they use self-hosted vs GH-hosted runners, but I see your point. 

@MarkLodato further thoughts?

Reply:
Yeah, perhaps this is better taken up elsewhere. Reading the docs again, I see that you're right that since we are signing the provenance in the reusable workflow ourselves (and not Github Actions itself) that the builder id should reflect the workflow's trust base, but more examples would make it easier to understand.

Reply:
Added code to generate the builder ID from the OIDC token